core/config/project_settings.cpp

```diff
@@ -344,7 +349,7 @@ void ProjectSettings::_get_property_list(List<PropertyInfo> *p_list) const {
 		vc.name = E.key;
 		vc.order = v->order;
 		vc.type = v->variant.get_type();
-		if (vc.name.begins_with("input/") || vc.name.begins_with("import/") || vc.name.begins_with("export/") || vc.name.begins_with("/remap") || vc.name.begins_with("/locale") || vc.name.begins_with("/autoload")) {
+		if (v->internal || vc.name.begins_with("input/") || vc.name.begins_with("import/") || vc.name.begins_with("autoload/") || vc.name.begins_with("editor_plugins/") || vc.name.begins_with("shader_globals/")) {
```

Comment:
Internal properties are for use only if there are single/few properties in the group that you want to hide. Meantime e.g. `input/` filters out whole input map; there is no single property that you'd want to edit outside the dedicated tab.

We could maybe add a way to define groups that should be hidden, but this code is in project settings, so it's probably ok like this.

Reply:
Well, it feels weird to have a system to mark properties internal that's only used for a couple properties but not for the many which are actually being explicitly skipped.

Why can't we call `set_as_internal` when registering input mappings or autoloads?

Reply:
I could do that. It's just less error prone to do it for the whole group if you know that none of the properties need to be visible.

Reply:
My assumption is that all those "groups" of properties are actually defined in `for` loops and thus marking them internal is simpler than filtering them later on:
```
diff --git a/core/config/project_settings.cpp b/core/config/project_settings.cpp
index ff5ff83bf8..b91b2b64da 100644
--- a/core/config/project_settings.cpp
+++ b/core/config/project_settings.cpp
@@ -1151,7 +1151,7 @@ void ProjectSettings::_add_builtin_input_map() {
 			action["events"] = events;
 
 			String action_name = "input/" + E.key;
-			GLOBAL_DEF(action_name, action);
+			GLOBAL_DEF_INTERNAL(action_name, action);
 			input_presets.push_back(action_name);
 		}
 	}
```

That being said, for default input actions for example, we do want them to appear in the docs, so it can be different from other cases.

Reply:
This loop unfortunately doesn't include user-defined actions. I made these internal, but the filter will still be there.
Same is with other properties filtered here. They are dynamic and user-created and I don't really see other way to hide them, as they are "created" with `set()`.